application/views/registry_instrument_view.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+import structlog
+from flask import Blueprint, jsonify, make_response
+
+from application.controllers.basic_auth import auth
+from application.controllers.registry_instrument import RegistryInstrument
+
+log = structlog.wrap_logger(logging.getLogger(__name__))
+
+registry_instrument_view = Blueprint("registry_instrument_view", __name__)
+
+
+@registry_instrument_view.before_request
+@auth.login_required
+def before_registry_instrument_view():
+    pass
+
+
+@registry_instrument_view.route("/registry-instrument/exercise-id/<exercise_id>", methods=["GET"])
+def get_registry_instruments(exercise_id):
+    registry_instruments = RegistryInstrument().get_registry_instruments_by_exercise_id(exercise_id)
+
+    if registry_instruments:
+        return make_response(jsonify(registry_instruments), 200)
+
+    return make_response("Not Found", 404)
```

Comment:
As per the comment in the API doc, I wonder if this would be better to return a `200` and an empty list `[ ]`. Its a valid scenario that there might not yet be any selected registry instruments for the given collection exercise. Returning a `404` feels heavy handed and makes the Rops usage a bit more complicated.

Comments welcomed?

Reply:
I can see an argument for both. When a new collection exercise is setup it would be expected and valid that no registry instruments are selected and that does feel heavy handed to be using a 404. Though is there a gap where we could return a 200 and an empty list unintentionally? I lean towards the 404 but i'm not sure how much more complicated Rops would be to implement then.

Reply:
Done